Add CAS and CAS library

hharchani opened the pull request:
Also, loaded the composer's autoload in root index.php

polybuildr commented:
We're using the imperative mood for commits. Could you please update? :P

nisargjhaveri commented on src/app/libraries/cas_lib.php:
```diff
@@ -0,0 +1,37 @@
+                $cfg['cas_port'],
+                $cfg['cas_context']
+            );
+            phpCAS::setNoCasServerValidation();
```
We might also want to give a certificate here.

hharchani replied:
might? toh jab denge tab add kar lenge, nahi?

polybuildr commented:
When you squash this commit, please put in the "fixes T39" trigger somewhere in your commit body so that we can link this with phab.

hharchani commented:
@nisargjhaveri Any thing left here? If not, you can merge it.

nisargjhaveri commented on src/app/libraries/cas_lib.php:
```diff
@@ -0,0 +1,42 @@
+            //phpCAS::setCasServerCACert($cfg['cas_server_ca_cert']);
+
+            // For quick testing, SSL validation of the CAS server can be disabled.
+            phpCAS::setNoCasServerValidation();
```
This should be set automatically only when cert file is not given.

nisargjhaveri commented:
And, squash, please?

polybuildr commented:
And in the squashed commit please use the phab triggers.

hharchani commented:
What's squash?

polybuildr commented:
http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git/

First two answers are very useful.

hharchani committed:
Add CAS using composer and add CAS lib

- Edit gulpfile to copy all pem files in src
- composer autoloader is setup
Fix T39

hharchani commented:
{Fix T39}

nisargjhaveri commented:
Looks fine.
